### PR TITLE
update genesis file content

### LINKED ISSUE #108: Add information regarding the genesis file
Link to the following regarding the genesis file creation:
https://consensys.net/blog/quorum/hyperledger-besu-how-to-create-an-ethereum-genesis-file/

Check if any of the information should be added to the documentation as well.

### PR DESCRIPTION
**Pull request by alexandratran**
## Pull Request Description

Update genesis file content for clarity and consistency with Besu docs (see https://github.com/hyperledger/besu-docs/pull/783).

## Fixed Issue(s)

fixes #108

### CODE REVIEW

File: docs/Reference/genesis.md [hunk 1]
```diff
@@ -1,48 +1,92 @@
...
+|---------------------|-:-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Milestone blocks    | [Milestone blocks for the network](#milestone-blocks).                                                                                                                                      |
+| `chainId`           | [Chain ID for the network](../Concepts/NetworkAndChainID.md).                                                                                                                               |
+| `istanbul`          | Specifies network uses [IBFT](../HowTo/Configure/Consensus-Protocols/IBFT.md) or [QBFT](../HowTo/Configure/Consensus-Protocols/QBFT.md) and contains [IBFT](../HowTo/Configure/Consensus-Protocols/IBFT.md#genesis-file) or [QBFT configuration items](../HowTo/Configure/Consensus-Protocols/QBFT.md#genesis-file), respectively. |
```
Comment (baptiste-b-pegasys) [inline docs/Reference/genesis.md:18]: I would move "configuration items" outside the link, so this term can apply to both IBFT and QBFT.
```suggestion
| `istanbul`          | Specifies network uses [IBFT](../HowTo/Configure/Consensus-Protocols/IBFT.md) or [QBFT](../HowTo/Configure/Consensus-Protocols/QBFT.md) and contains [IBFT](../HowTo/Configure/Consensus-Protocols/IBFT.md#genesis-file) or [QBFT](../HowTo/Configure/Consensus-Protocols/QBFT.md#genesis-file)  configuration items, respectively. |
```

File: docs/Reference/genesis.md [hunk 2]
```diff
@@ -1,48 +1,92 @@
...
+
+* `difficulty`.
+* `extraData`.
+* `mixHash`.
+
+The following table describes the genesis block parameters with the same purpose across all
+consensus protocols.
+
+| Item                | Description                                                                                                                             |
+|---------------------|-:---------------------------------------------------------------------------------------------------------------------------------------|
```
Comment (baptiste-b-pegasys) [inline docs/Reference/genesis.md:43]: Does the `:` has a reason?
```suggestion
|---------------------|-----------------------------------------------------------------------------------------------------------------------------------------|
```

File: docs/Reference/genesis.md [hunk 3]
```diff
@@ -1,48 +1,92 @@
...
+| `clique`            | Specifies network uses [Clique](../HowTo/Configure/Consensus-Protocols/Clique.md) and contains Clique configuration items.                                                                  |
+| `txnSizeLimit`      | Maximum transaction size. The default is `64` (kilobytes), increased from Ethereum's default `32`. This is configurable up to `128`.                                                        |
+| `maxCodeSize`       | Maximum smart contract code size. The default is `32` (kilobytes), increased from Ethereum's default `24`. This is configurable up to `128`.                                                |
+| `isQuorum`          | Set to `true` to allow interoperable private transactions between GoQuorum and [Hyperledger Besu](https://besu.hyperledger.org/en/stable/) using the Tessera private transaction manager.   |
```
Comment (ricardolyn) [inline docs/Reference/genesis.md:22]: `isQuorum` is more than that. it actually enables `geth` to work as GoQuorum and do some checks like make sure the gas fee is zero. for ex, you can use `isQuorum` to use the GoQuorum engine without using Tessera.

File: docs/Reference/genesis.md (hunk 2 above)
Comment (NicolasMassart) [reply docs/Reference/genesis.md:43]: `:` is [used to align text in cells at rendering](https://www.markdownguide.org/extended-syntax/#alignment)

File: docs/Reference/genesis.md [hunk 4]
```diff
@@ -1,48 +1,92 @@
...
-    "0x0fbdc686b912d7722dc86510934589e0aaf3b55a": {
-      "balance": "1000000000000000000000000000"
-    },
-    "0x9186eb3d20cbd1f5f992a950d808c4495153abd5": {
-      "balance": "1000000000000000000000000000"
-    },
-    "0x0638e1574728b6d862dd5d3a3e0942c3be47d996": {
-      "balance": "1000000000000000000000000000"
+---
+description: Configuration items specified in the Hyperledger Besu genesis file
```
Comment (macfarla) [inline docs/Reference/genesis.md:2]: s/Besu/GoQuorum

File: docs/Reference/genesis.md [hunk 5]
```diff
@@ -1,48 +1,92 @@
...
+| `isQuorum`          | Set to `true` to enable `geth` to work as GoQuorum and make additional checks, for example, ensure the gas fee is zero.                                                                     |
+
+!!! note
+
+    Using the [Raft consensus protocol](../HowTo/Configure/Consensus-Protocols/Raft.md) doesn't require a specific item in the `config` object.
+
+## Genesis block parameters
+
+The purpose of some genesis block parameters varies depending on the consensus protocol ([IBFT](../HowTo/Configure/Consensus-Protocols/IBFT.md),
+[QBFT](../HowTo/Configure/Consensus-Protocols/QBFT.md), [Raft](../HowTo/Configure/Consensus-Protocols/Raft.md),or
```
Comment (baptiste-b-pegasys) [inline docs/Reference/genesis.md:31]:
```suggestion
[QBFT](../HowTo/Configure/Consensus-Protocols/QBFT.md), [Raft](../HowTo/Configure/Consensus-Protocols/Raft.md), or
```

File: docs/HowTo/Configure/Contracts-in-Genesis.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+To pre-deploy contracts when starting GoQuorum, specify the contract code in the [genesis file](GenesisOptions.md).
+
+!!! example "Contract code in the genesis file"
+
+    ```json
+    {
+      ...
+      "alloc": {
+          "0x0ffd23af8eebc60b3cfdeed6f814988757237314": {
```
Comment (antonydenyer) [inline docs/HowTo/Configure/Contracts-in-Genesis.md:15]: Nit, fix indentation.

File: docs/HowTo/Configure/Contracts-in-Genesis.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+      },
+      ...
+    }
+    ```
+
+The contract code in the genesis file defines the:
+
+* Address.
+* Balance.
+* Bytecode.
```
Comment (antonydenyer) [inline docs/HowTo/Configure/Contracts-in-Genesis.md:32]: Note! the bytecode is the binary runtime code and not the code to deploy the contract. You can generate this using `solc --bin-runtime contract.sol`

**Review by antonydenyer — APPROVED** [review]
Couple of suggestions, but no blockers

File: docs/HowTo/Configure/Contracts-in-Genesis.md (hunk 6 above)
Comment (NicolasMassart) [reply docs/HowTo/Configure/Contracts-in-Genesis.md:15]: you could try this tool: https://jsonformatter.curiousconcept.com/

```json
{
   ...
   "alloc":{
      "0x0ffd23af8eebc60b3cfdeed6f814988757237314":{
         "balance":"0x100000000000000000000000000000000000000000000000000",
         "code":"0x6080604052600436106043576000357c010000000000000000000000000000000000000000000000000000000090048063010fc84214604857806355241077146070575b600080fd5b348015605357600080fd5b50605a60a7565b6040518082815260200191505060405180910390f35b348015607b57600080fd5b5060a560048036036020811015609057600080fd5b810190808035906020019092919050505060ad565b005b60005481565b80600081905550807f04474795f5b996ff80cb47c148d4c5ccdbe09ef27551820caa9c2f8ed149cce360405160405180910390a25056fea165627a7a7230582038cb7ea327af8f73feabcfbff64498f1e74831e67f7c75286760d3845c6747c70029",
         "storage":{
            "7aa07e0c924147697605046b7c2c32645b7bbfb41e0ac5d0a84ac93cbb759798":"0000000000000000000000000000000000000000000000000000000000000001",
            "cea2b0602db61f92b76ec4402875cc38eedc9fc425cb1b697fc2265d50fc20fb":"0000000000000000000000000000000000000000000000000000000000000001"
         }
      }
   },
   ...
}
```

**Review by NicolasMassart — APPROVED** [review]
Looks good to me once the few fixes indicated by @antonydenyer are made

**Review by rolandtyler — APPROVED** [review]
LGTM

**Pull request merged**